File: frontend/src/lib/api/generation.ts
import { apiClient } from './client';
import { GenerationRequest, GenerationPreviewResponse } from '@/types';

/**
 * 1行説明からシナリオを自動生成する。
 * キャラクターは生成対象外 (手動で別途登録する)。
 */
export async function generateScenario(
  data: GenerationRequest
): Promise<GenerationPreviewResponse> {
  return apiClient.post<GenerationPreviewResponse>(
    '/api/v1/generation/scenario',
    data
  );
}

File: backend/app/api/v1/generation.py
from fastapi import APIRouter

from app.schemas.generation import GenerationPreviewResponse, GenerationRequest
from app.services.generation import GenerationService

router = APIRouter()


@router.post("", response_model=GenerationPreviewResponse)
async def generate_scenario(
    request: GenerationRequest,
) -> GenerationPreviewResponse:
    """
    1行説明からシナリオを自動生成する。

    - **description**: 1〜500文字の説明文
    - **locale**: 言語設定 (ja/en)

    キャラクターは生成対象外。キャラクターは従来通り手動で登録・選択する。
    """
    service = GenerationService()
    return await service.generate(request)
